Experimental push notification support (via NTFY)

mgrimace opened the pull request:
This PR adds experimental push notification support via [NTFY](https://ntfy.sh/), closing FR #90 

Summary of changes:
- users input NTFY server, topic, and optionally auth in the `config.json`, parsed using `Config.ts`
- `Logger.ts` matches [LOG], [ERROR], and [WARN] against customizable key words to generate notifications (with comments added to show where such customization can occur)
- `Ntfy.sh` sends customizable notifications using `Axios.ts`, which include emojis and prioritization based on type (e.g., high priority for warnings).
-  `Readme` updated to briefly describe setup, and customization options. 

By default, notifications are sent for each account: script started, 2FA/passwordless code, script successfully completed, and major warnings (e.g., searches not completed).

Huge thanks to @AariaX for fixing axios support for sending.

AariaX commented on src/util/Logger.ts:
```diff
@@ -1,24 +1,43 @@
+        log: [
+            message.toLowerCase().includes('started tasks for account'),
+            message.toLowerCase().includes('press the number'),
+            message.toLowerCase().includes('this script collected'),
```
```suggestion
            message.toLowerCase().includes('the script collected'),
```
well maybe this is why

AariaX commented:
https://github.com/TheNetsky/Microsoft-Rewards-Script/blob/2a1a90c937c3a8a5c44b57335dd8df458034f369/src/index.ts#L100

Changing this to "Completed tasks for ALL accounts" might be nice?

mgrimace commented:
> https://github.com/TheNetsky/Microsoft-Rewards-Script/blob/2a1a90c937c3a8a5c44b57335dd8df458034f369/src/index.ts#L100
> 
> Changing this to "Completed tasks for ALL accounts" might be nice?

I do get a [LOG] line that says 'completed tasks for ALL accounts' for each cluster/account = 2 notifications. I do personally prefer having it 'completed for email' instead so I could verify both completed.

If this change would generate one summary notification for all clusters that is beneficial. If it shows up as a [WARN] it would be easier to isolate from logs, but as a notification it might also (at a glance) look like there's a problem if it does show up with the warn tag.

Does all workers destroyed == tasks completed? Then there'd be no harm in changing it.

TheNetsky commented:
Since this is still WIP, I'll publish 1.5 without this being added for now. Since I think 1.5 is stable enough to be set as the main release.

mgrimace commented:
> Since this is still WIP, I'll publish 1.5 without this being added for now. Since I think 1.5 is stable enough to be set as the main release.

Of course, makes sense 1.5 has been absolutely stable for me, and this is a 'bonus' feature IMO. As far as I can tell I have no further work on NTFY, it is working as intended for me. I added how to customize/add/remove alerts in the readme (and comments in the logger.ts) in case any new good keyword alerts come up. 

As for review, I'd suggest a review of index.ts to make sure that's current with the two necessary `await`s that I added. Otherwise it's fit to ship when you're ready (from my end).

mgrimace commented:
Would it be easier if I close this PR and do a new one with 1.5 in main now?
